src/redux/missions/MissionsSlice.js

```diff
@@ -0,0 +1,59 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
+
+const BASE_URL = 'https://api.spacexdata.com/v3/missions';
+
+const initialState = {
+  missionstore: [],
+  status: 'idle',
+  error: null,
+};
+export const fetchMissions = createAsyncThunk('missions/fetchMissions', async () => {
+  try {
+    const response = await axios.get(BASE_URL);
+    return response.data;
+  } catch (error) {
+    return error.message;
+  }
+});
+export const missionsSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    joinMission: (state, payload) => ({
+      ...state,
+      missionstore: state.missionstore.map((mission) => {
+        if (mission.mission_id === payload.payload) {
+          if (mission.reserved === true) {
+            return { ...mission, reserved: false };
+          }
+          return { ...mission, reserved: true };
+        }
+        return mission;
+      }),
+    }),
+  },
```

Comment:
- Optional
I strongly recommend having a separate reducer for leaving the mission. It is always recommended to have one reducer per action.